pkg/statistics/handle/autoanalyze/internal/heap/heap.go

```diff
@@ -0,0 +1,306 @@
+// Copyright 2017 The Kubernetes Authors.
+// Copyright 2024 PingCAP, Inc.
```

Comment:
Better to add some comments to explain why we copy this implementation instead of using other third-party libs.